### PR TITLE
New Cheat Sheet : Tcpdump

### PR DESCRIPTION
**Pull request by vduddu**
## Description of new Instant Answer, or changes

Added a new cheat sheet for basic and advanced tcpdump usage 

## Related Issues and Discussions

N/A

## People to notify

@vduddu

---

Instant Answer Page: https://duck.co/ia/view/tcpdump_cheat_sheet

---

### CODE REVIEW

**Comment by daxtheduck** [comment]
@vduddu Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/tcpdump.json [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    "name": "Tcpdump",
+    "description": "Package analyser and sniffing tool",
+
+    "metadata": {
+        "sourceName": "Tcpdump Command Examples",
+        "sourceUrl" : "http://www.thegeekstuff.com/2010/08/tcpdump-command-examples",
+	"sourceName": "Tcpdump tutorial and Primer",
+        "sourceUrl" : "https://danielmiessler.com/study/tcpdump/#gs.SJSyNc8",
+	"sourceName": "Advanced Tcpdump filtering",
+        "sourceUrl" : "https://www.wains.be/pub/networking/tcpdump_advanced_filters.txt"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/tcpdump.json]: Lets keep one source url and corresponding source name. And remove others

File: share/goodie/cheat_sheets/json/tcpdump.json [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+                "key": "tcpdump 'tcp\\[13\\] & 32!=0' -i  \\[interface\\]",
+                "val": "Captures tcp packets with URG flag set"
+            },
+            {
+                "key": "tcpdump 'tcp\\[13\\] & 16!=0' -i \\[interface\\]",
+                "val": "Captures TCP packets with ACK flag set"
+            },
+	    {
+                "key": "tcpdump 'tcp\\[13\\] & 8!=0' -i  \\[interface\\]",
+                "val": "Captures Tcp packets with PSH flag set"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/tcpdump.json]: Use one representation for TCP. Now you have tcp,Tcp and TCP.

File: share/goodie/cheat_sheets/json/tcpdump.json [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+                "val": "some bits are set when applying mask z to protocol\\[x:y\\]"
+    },
+	{
+                "key": "protocol\\[x:y\\] & z = z",
+                "val": "every bits are set to z when applying mask z to protocol\\[x:y\\]"
+    },
+	{
+                "key": "protocol\\[x:y\\] = z",
+                "val": "protcol\\[x:y\\] has exactly the bits set to z"
+    }
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/tcpdump.json]: Start the above val fields with capital letters

Comment (vduddu) [inline share/goodie/cheat_sheets/json/tcpdump.json]: Done :)

File: share/goodie/cheat_sheets/json/tcpdump.json (hunk 2 above)
Comment (vduddu) [inline share/goodie/cheat_sheets/json/tcpdump.json]: Done :)

File: share/goodie/cheat_sheets/json/tcpdump.json (hunk 1 above)
Comment (vduddu) [inline share/goodie/cheat_sheets/json/tcpdump.json]: Done :)

File: share/goodie/cheat_sheets/json/tcpdump.json [hunk 4]
```diff
@@ -5,11 +5,7 @@
 
     "metadata": {
         "sourceName": "Tcpdump Command Examples",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/tcpdump.json]: Change source name to "The Geek Stuff"

File: share/goodie/cheat_sheets/json/tcpdump.json [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+                "key": "tcpdump -w \\[output file .pcap\\] -i \\[interface\\]",
+                "val": "Saves the output into a .pcap file "
+            },
+	    {
+                "key": "tcpdump -r \\[output file .pcap\\]",
+                "val": "Reads the data from a .pcap file "
+            },
+	    {
+                "key": "tcpdump -n -i \\[interface\\]",
+                "val": "Prints the IP addresses instead of DNS addresses "
```
Comment (Hasil-Sharma) [inline share/goodie/cheat_sheets/json/tcpdump.json]: Chuck trailing spaces

**Comment by Hasil-Sharma** [comment]
@vduddu : One last thing from my side. Just run your json by a beautifier/validator to indent your json properly. Everything else looks good to me 😄

**Comment by pjhampton** [comment]
This PR looks like it has went stale. Closing 

If you are looking for something more substantial to work through that will have a big impact for the users of DuckDuckGo, please check out the [DuckDuckHack Issues Table](https://duckduckhack.com/issues). 🚀  😄

**Pull request closed without merging**